### PR TITLE
store/tikv: fix a potential goroutine leak problem in copIterator

### PR DESCRIPTION
**Pull request by tiancaiamao**
copIterator.Close() closes finished channel and wait for worker goroutines to
exit, worker goroutine send response to channel but nobody receive from the
channel (copIterator already closed), so those goroutine leak and Close() function
hang forever.

Change the worker goroutine to check both response channel and finished channel.

@zz-jason @coocood @shenli

### CODE REVIEW

**Comment by tiancaiamao** [comment]
@coocood @zz-jason PTAL

File: executor/executor_test.go [hunk 1]
```diff
@@ -2460,14 +2461,24 @@ func (s *testSuite) TestEarlyClose(c *C) {
...
+		c.Assert(err1, IsNil)
 		rs := rss[0]
 		chk := rs.NewChunk()
 		err = rs.NextChunk(ctx, chk)
 		c.Assert(err, IsNil)
 		rs.Close()
 	}
+
+	// Goroutine should not leak when error happen.
+	gofail.Enable("github.com/pingcap/tidb/store/tikv/handleTaskOnceError", `return(true)`)
```
Comment (winkyao) [inline executor/executor_test.go]: add `defer testleak.AfterTest(c)()` to detect leak?

Comment (tiancaiamao) [reply executor/executor_test.go]: It's done in `testsuite.SetUpSuite()` and `TearDownSuite()` @winkyao

File: executor/executor_test.go [hunk 2]
```diff
@@ -2460,14 +2461,24 @@ func (s *testSuite) TestEarlyClose(c *C) {
...
+	defer gofail.Disable("github.com/pingcap/tidb/store/tikv/handleTaskOnceError")
+	rss, err := tk.Se.Execute(ctx, "select * from earlyclose")
+	c.Assert(err, IsNil)
+	rs := rss[0]
+	chk := rs.NewChunk()
+	rs.NextChunk(ctx, chk)
```
Comment (shenli) [inline executor/executor_test.go]: Please get all the result in a loop.

**Comment by shenli** [comment]
LGTM

**Pull request merged**